### PR TITLE
[FIX] website: navigation in duplicated gallery snippet

### PR DESCRIPTION
**Pull request by bvr-odoo**
Fix in gallery snippet (carousel mode). Create a new id and update
links id when this snippet is duplicated.

task-2162952

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/static/src/js/editor/snippets.options.js [hunk 1]
```diff
@@ -570,6 +570,15 @@ options.registry.gallery = options.Class.extend({
         this.$target.find('.carousel').attr('id', 'slideshow_' + uuid);
         this.$target.find('[data-target]').attr('data-target', '#slideshow_' + uuid);
     },
+    /**
+     * @override
+     */
+    onClone: function () {
+        var uuid = new Date().getTime();
+        this.$target.find('.carousel').attr('id', 'slideshow_' + uuid);
+        this.$target.find('[data-target]').attr('data-target', '#slideshow_' + uuid);
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js]: Seems too generic as selector, `[data-target]` could be used by many other things in the future. Can't you find something more precise like you did with `[data-slide]` ?

File: addons/website/static/src/js/editor/snippets.options.js [hunk 2]
```diff
@@ -570,6 +570,15 @@ options.registry.gallery = options.Class.extend({
         this.$target.find('.carousel').attr('id', 'slideshow_' + uuid);
         this.$target.find('[data-target]').attr('data-target', '#slideshow_' + uuid);
     },
+    /**
+     * @override
+     */
+    onClone: function () {
+        var uuid = new Date().getTime();
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js]: Is not this equal to `Date.now()` ?

File: addons/website/static/src/js/editor/snippets.options.js [hunk 3]
```diff
@@ -698,10 +698,10 @@ options.registry.gallery = options.Class.extend({
      */
     mode: function (previewMode, value, $li) {
         this.$target.css('height', '');
-        this[value]();
         this.$target
             .removeClass('o_nomode o_masonry o_grid o_slideshow')
             .addClass('o_' + value);
+        this[value]();
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js]: Review the commit message. Don't understand the "and we needed to do it" part :wink:

**Comment by qsm-odoo** [comment]
Same here

File: addons/website/static/src/js/editor/snippets.options.js (hunk 2 above)
Comment (bvr-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: yes

**Pull request closed without merging**